google-cloud-storage/lib/google/cloud/storage/hmac_key/list.rb

```diff
@@ -0,0 +1,182 @@
+          # Retrieves remaining results by repeatedly invoking {#next} until
+          # {#next?} returns `false`. Calls the given block once for each
+          # result, which is passed as the argument to the block.
+          #
+          # An Enumerator is returned if no block is given.
+          #
+          # This method will make repeated API calls until all remaining results
+          # are retrieved. (Unlike `#each`, for example, which merely iterates
+          # over the results returned by a single API call.) Use with caution.
+          #
+          # @param [Integer] request_limit The upper limit of API requests to
+          #   make to load all HMAC keys. Default is no limit.
+          # @yield [key] The block for accessing each HMAC key.
+          # @yieldparam [HmacKey] key The HMAC key object.
+          #
+          # @return [Enumerator]
+          #
+          # @example Iterating each HMAC key by passing a block:
+          #   require "google/cloud/storage"
+          #
+          #   storage = Google::Cloud::Storage.new
+          #
+          #   hmac_keys = storage.hmac_keys
+          #   hmac_keys.all do |key|
+          #     puts key.access_id
+          #   end
+          #
+          # @example Using the enumerator by not passing a block:
+          #   require "google/cloud/storage"
+          #
+          #   storage = Google::Cloud::Storage.new
+          #
+          #   hmac_keys = storage.hmac_keys
+          #   all_names = hmac_keys.all.map do |key|
+          #     key.access_id
+          #   end
+          #
+          # @example Limit the number of API calls made:
+          #   require "google/cloud/storage"
+          #
+          #   storage = Google::Cloud::Storage.new
+          #
+          #   hmac_keys = storage.hmac_keys
+          #   hmac_keys.all(request_limit: 10) do |key|
+          #     puts key.access_id
+          #   end
+          #
+          def all request_limit: nil
+            request_limit = request_limit.to_i if request_limit
+            unless block_given?
+              return enum_for :all, request_limit: request_limit
+            end
+            results = self
+            loop do
+              results.each { |r| yield r }
+              if request_limit
+                request_limit -= 1
+                break if request_limit < 0
+              end
+              break unless results.next?
+              results = results.next
+            end
+          end
+
+          ##
+          # @private New HmacKey::List from a Google API Client
+          # Google::Apis::StorageV1::HmacKeysMetadata object.
+          def self.from_gapi gapi_list, service, service_account_email: nil,
+                             show_deleted_keys: nil, max: nil, user_project: nil
+            hmac_keys = new(Array(gapi_list.items).map do |gapi_object|
+              HmacKey.from_gapi_metadata gapi_object, service,
+                                         user_project: user_project
```

Comment:
Set `project_id` value on HmacKey instance.

Reply:
The `gapi_object` variable here holds the `Google::Apis::StorageV1::HmacKeyMetadata` instance returned by the RPC and will be held in the `HmacKey` instance, so that's OK isn't it?